x/pylons/keeper/msg_server_execute_recipe_test.go

```diff
@@ -1,82 +1,417 @@
+			valid:                 false,
+		},
+		{
+			decs:                "Invalid coinInputs index",
+			cookbook:            *cookbook,
+			unEnabledCookbook:   false,
+			changeIdCookbook:    false,
+			recipe:              *recipe,
+			changeIdRecipe:      false,
+			updateCoinInputs:    true,
+			updateEntriesRecipe: false,
+			execution: types.MsgExecuteRecipe{
+				Creator:         types.TestCreator,
+				CookbookId:      "",
+				RecipeId:        "",
+				CoinInputsIndex: 2,
+				ItemIds:         nil,
+			},
+			changeExcutionCreator: false,
+			valid:                 false,
+		},
+		{
+			decs:                "Invalid PaymentInfo",
+			cookbook:            *cookbook,
+			unEnabledCookbook:   false,
+			changeIdCookbook:    false,
+			recipe:              *recipe,
+			changeIdRecipe:      false,
+			updateCoinInputs:    false,
+			updateEntriesRecipe: false,
+			execution: types.MsgExecuteRecipe{
+				Creator:         types.TestCreator,
+				CookbookId:      "",
+				RecipeId:        "",
+				CoinInputsIndex: 0,
+				ItemIds:         nil,
+				PaymentInfos: []types.PaymentInfo{
+					{
+						PurchaseId:    "1",
+						ProcessorName: "test",
+						PayerAddr:     "pylon0123",
+						Amount:        sdk.OneInt(),
+						ProductId:     "1",
+						Signature:     "test",
+					},
+				},
+			},
+			changeExcutionCreator: false,
+			valid:                 false,
+		},
+		{
+			decs:                "Amount minted reached maximum limit",
+			cookbook:            *cookbook,
+			unEnabledCookbook:   false,
+			changeIdCookbook:    false,
+			recipe:              *recipe,
+			changeIdRecipe:      false,
+			updateCoinInputs:    false,
+			updateEntriesRecipe: true,
+			execution: types.MsgExecuteRecipe{
+				Creator:         types.TestCreator,
+				CookbookId:      "",
+				RecipeId:        "",
+				CoinInputsIndex: 0,
+				ItemIds:         nil,
+			},
+			changeExcutionCreator: false,
+			valid:                 false,
+		},
+		{
+			decs:                "Process PaymentInfo error",
+			cookbook:            *cookbook,
+			unEnabledCookbook:   false,
+			changeIdCookbook:    false,
+			recipe:              *recipe,
+			changeIdRecipe:      false,
+			updateCoinInputs:    false,
+			updateEntriesRecipe: true,
+			execution: types.MsgExecuteRecipe{
+				Creator:         types.TestCreator,
+				CookbookId:      "",
+				RecipeId:        "",
+				CoinInputsIndex: 0,
+				ItemIds:         nil,
+				PaymentInfos: []types.PaymentInfo{{
+					PurchaseId:    "test",
+					ProcessorName: "TestPayment",
+					PayerAddr:     types.GenTestBech32FromString(types.TestCreator),
+					Amount:        sdk.NewInt(2),
+					ProductId:     "testProductId",
+					Signature:     genTestPaymentInfoSignature("testPurchaseId", types.GenTestBech32FromString(types.TestCreator), "testProductId", sdk.NewInt(2), privKey),
+				}},
+			},
+			changeExcutionCreator: false,
+			valid:                 false,
+		},
+		{
+			decs:                "Error lock coins for execution",
```

Comment:
"Error locking coins for execution" might be a better error message.

Reply:
thanks, I have updated this